test/acceptance/test/core_operations_test.go

```diff
@@ -60,180 +61,310 @@ spec:
-	require.True(t, found)
-	c, err := github.NewClient(github.WithOAuth2Token(token), github.WithDestructiveAPICalls(true), github.WithConditionalRequests(true))
-	require.NoError(t, err)
-	client = c
-	log.Info("Ensuring wego repo does not exist...")
-	ensureWegoRepoIsAbsent(t)
-	log.Info("Ensuring flux version is set...")
-	ensureFluxVersion(t)
-	log.Info("Checking initial status...")
-	checkInitialStatus(t)
-	log.Info("Install flux...")
-	installFlux(t)
-	log.Info("Setting up test repository...")
-	setUpTestRepo(t) // create repo with simple nginx manifest
-	defer deleteRepos(t)
-	log.Info("Adding test repository to cluster...")
-	require.NoError(t, err)
-	addRepo(t) // add new repo to cluster
-	log.Info("Waiting for workload to start...")
-	waitForNginxDeployment(t)
-}
+	BeforeEach(func() {
+		By("Given I have a wego binary installed on my local machine", func() {
+			Expect(FileExists(WEGO_BIN_PATH)).To(BeTrue())
+		})
 
-func addRepo(t *testing.T) {
-	keyFilePath := filepath.Join(os.Getenv("HOME"), ".ssh", "id_rsa")
-	if _, err := os.Stat(keyFilePath); os.IsNotExist(err) {
-		key := os.Getenv("GITHUB_KEY")
-		tmpFile, err := ioutil.TempFile("", "keyfile")
-		require.NoError(t, err)
-		defer tmpFile.Close()
-		require.NoError(t, ioutil.WriteFile(tmpFile.Name(), []byte(key), 600))
-		keyFilePath = tmpFile.Name()
-	}
+		By("Setup test", func() {
+			Expect(checkInitialStatus()).Should(Succeed())
+			Expect(setupTest()).Should(Succeed())
+			Expect(ensureWegoRepoIsAbsent()).Should(Succeed())
+			Expect(ensureFluxVersion()).Should(Succeed())
+			Expect(installFlux()).Should(Succeed())
+			Expect(setUpTestRepo()).Should(Succeed())
+		})
 
-	dir, err := os.Getwd()
-	require.NoError(t, err)
-	require.NoError(t, os.Chdir(tmpDir))
-	defer func() {
-		require.NoError(t, os.Chdir(dir))
-	}()
+	})
+
+	It("Verify add private repo when repo does not already exist", func() {
+
+		By("When i run 'wego add .'", func() {
+			dir, err := os.Getwd()
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(os.Chdir(tmpDir)).Should(Succeed())
+			defer Expect(os.Chdir(dir)).Should(Succeed())
+			command := exec.Command(WEGO_BIN_PATH, "add", ".")
+			session, err = gexec.Start(command, GinkgoWriter, GinkgoWriter)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		By("Then a private repo with name foo-cluster-wego is created on the remote git", func() {
+			Expect(ensureWegoRepoExists()).Should(Succeed())
+		})
 
-	cmdimpl.Add([]string{"."}, cmdimpl.AddParamSet{Name: "", Url: "", Path: "./", Branch: "main", PrivateKey: keyFilePath, Namespace: "wego-system", DeploymentType: "kustomize"})
+		By("kubectl get pods -n wego-system should list the source and kustomize controllers", func() {
+			Expect(waitForNginxDeployment()).Should(Succeed())
+			Expect(runCommandForGinkgo("kubectl get pods -n wego-system")).Should(Succeed())
+			Eventually(session).Should(gbytes.Say("kustomize-controller"))
+			Eventually(session).Should(gbytes.Say("source-controller"))
```

Comment:
When you use `gbytes.Say`, it implicitly also verify the order of those controllers in the output which is not important in this case. I would suggest using the `ContainsSubstring` or `MatchRegexp` instead.